pstress/repeat_crash.sh

```diff
@@ -0,0 +1,227 @@
+############################################################################################
+# Created by Mohit Joshi, Percona LLC                                                      #
+# Creation date: 07-JAN-2021                                                               #
+#                                                                                          #
+# This script is created to repeat crashes found during pstress runs.                      #
+# The script requires the incident directory, the mysql basedir, mode and the nth step     #
+# from where the pstress runs will be resumed to repeat crashes.                           #
+# The script requires to be run from <source>/pstress directory                            #
+# Usage: ./repeat_crash.sh -i$HOME/path/to/incident/dir -b/path/to/basedir -s3 -mcurrent   #
+############################################################################################
+
+#!/bin/bash
+
+Help()
+{
+echo "Syntax: repeat_crash [-i|s|m|b]";
+echo "options:
+-i	Path to the Incident Directory where pstress was executed
+-s	Nth step where the crash is seen
+-m	Select the mode: previous|current|after
+        previous=> Resume pstress from (Nth-2) step
+        current => Resume pstress from (Nth-1) step
+        after   => Resume pstress from (Nth) step
+-b      Path to the mysql base directory"
+}
+
+if [ $# -eq 0 ]
+then
+  Help
+  exit 1
+fi
+
+while getopts i:s:m:b: flag
+do
+  case "${flag}" in
+    i) incident_directory=${OPTARG};;
+    s) step=${OPTARG};;
+    m) mode=${OPTARG};;
+    b) basedir=${OPTARG};;
+  esac
+done
+
+# Output Function
+echoit() {
+  echo "[$(date +'%T')] $1"
+}
+
+echoit "========================================================="
+echoit "Incident Directory: $incident_directory";
+echoit "Nth step: $step";
+echoit "Mode previous|current|after: $mode";
+echoit "Base directory: $basedir";
+echoit "========================================================="
+
+RANDOM=`date +%s%N | cut -b14-19`; RANDOMD=$(echo $RANDOM$RANDOM$RANDOM | sed 's/..\(......\).*/\1/')
+mkdir $RANDOMD
+workdir=`pwd`/$RANDOMD
+SCRIPT_PWD=$(cd `dirname $0` && pwd)
+CONFIGURATION_FILE=pstress-run.conf
+source $incident_directory/$CONFIGURATION_FILE
+SEED=$(<$incident_directory/seed)
+
+
+if [ ! -d "$incident_directory" ]; then
```

Comment:
put this check before source $incident_directory/$CONFIGURATION_FILE

Reply:
Good point. I'll add it